lib/tapioca/gem/listeners/methods.rb

```diff
@@ -73,15 +73,29 @@ def compile_method(tree, symbol_name, constant, method, visibility = RBI::Public
           return unless method_owned_by_constant?(method, constant)
           return if @pipeline.symbol_in_payload?(symbol_name) && !@pipeline.method_in_gem?(method)
 
+          is_alias = method.name != method.original_name
           signature = lookup_signature_of(method)
-          method = T.let(signature.method, UnboundMethod) if signature
+          parameters = T.let(nil, T.nilable(T::Array[[Symbol, T.nilable(Symbol)]]))
+
+          if signature
+            method = T.let(signature.method, UnboundMethod)
+          elsif is_alias && signature.nil? && constant.method_defined?(method.original_name)
+            alias_source_method = constant.instance_method(method.original_name)
+            signature = lookup_signature_of(alias_source_method)
+
+            # Skip abstract methods if they are defined this way.
+            # Aliasing abstract methods is likely to yield unwanted results
+            signature = nil if signature&.mode == "abstract"
+            parameters = T.let(signature&.method&.parameters, T.nilable(T::Array[[Symbol, T.nilable(Symbol)]]))
```

Comment:
```suggestion
            parameters = signature&.method&.parameters
```
I don't think we need this

Reply:
I see. We don't enforce `strong` so it's fine if these are `T.untyped` but given we iterate over it types are useful.